### PR TITLE
[libnick] Update to 2024.6.8

### PR DESCRIPTION
**Pull request by nlogozzo**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

**Adds support for macOS**

### CODE REVIEW

File: ports/libnick/vcpkg.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
   "license": "GPL-3.0",
-  "supports": "(windows & x64) | (linux & x64)",
+  "supports": "windows | linux | osx",
   "dependencies": [
     "curl",
     "gettext-libintl",
     {
       "name": "glib",
-      "platform": "linux"
+      "platform": "linux | osx"
```
Comment (BillyONeal) [inline ports/libnick/vcpkg.json:15]: I verified that the source says `elseif(LINUX)` and `elseif(APPLE)` rather than `NOT Win32` so this is correct.

**Comment by BillyONeal** [comment]
```console
C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1438~1.331\bin\Hostx64\x86\cl.exe   /TP -DSQLITE_HAS_CODEC -ID:\b\libnick\src\2024.6.7-c6f72ebccd.clean\include -external:ID:\installed\x86-windows\include -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  -std:c++20 -MDd /W4 /showIncludes /FoCMakeFiles\libnick.dir\src\network\networkmonitor.cpp.obj /FdCMakeFiles\libnick.dir\libnick.pdb /FS -c D:\b\libnick\src\2024.6.7-c6f72ebccd.clean\src\network\networkmonitor.cpp
cl : Command line warning D9025 : overriding '/W3' with '/W4'
D:\b\libnick\src\2024.6.7-c6f72ebccd.clean\include\network/networkmonitor.h(45): error C2695: 'Nickvision::Network::NetworkMonitor::AddRef': overriding virtual function differs from 'IUnknown::AddRef' only by calling convention
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\unknwnbase.h(125): note: see declaration of 'IUnknown::AddRef'
D:\b\libnick\src\2024.6.7-c6f72ebccd.clean\include\network/networkmonitor.h(46): error C2695: 'Nickvision::Network::NetworkMonitor::Release': overriding virtual function differs from 'IUnknown::Release' only by calling convention
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\unknwnbase.h(127): note: see declaration of 'IUnknown::Release'
D:\b\libnick\src\2024.6.7-c6f72ebccd.clean\include\network/networkmonitor.h(47): error C2695: 'Nickvision::Network::NetworkMonitor::QueryInterface': overriding virtual function differs from 'IUnknown::QueryInterface' only by calling convention
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\unknwnbase.h(121): note: see declaration of 'IUnknown::QueryInterface'
D:\b\libnick\src\2024.6.7-c6f72ebccd.clean\include\network/networkmonitor.h(48): error C2695: 'Nickvision::Network::NetworkMonitor::ConnectivityChanged': overriding virtual function differs from 'INetworkListManagerEvents::ConnectivityChanged' only by calling convention
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\netlistmgr.h(563): note: see declaration of 'INetworkListManagerEvents::ConnectivityChanged'
```

It looks like this is missing a `STDMETHODCALLTYPE`?

**Comment by BillyONeal** [comment]
Since you are upstream rather than pushing a patch to fix it here I submitted a PR directly

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update! LGTM as soon as build passes

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**